Graphics_Engine/Query/Color.cs

```diff
@@ -0,0 +1,93 @@
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Geometry;
+using BH.oM.Graphics;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Drawing;
+using System.Linq;
+
+namespace BH.Engine.Graphics
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****           Public Methods                  ****/
+        /***************************************************/
+
+        [Description("Gets a color from a gradient at the spacified position")]
+        public static Color Color(this Gradient gradient, double val)
+        {
+            // Find adjacent markers
+            KeyValuePair<decimal, Color> upper = gradient.Markers.FirstOrDefault(x => x.Key > (decimal)val);
+            KeyValuePair<decimal, Color> lower = gradient.Markers.LastOrDefault(x => x.Key < (decimal)val);
+
+            // Check adjacency on both sides
+            if (lower.Equals(default(KeyValuePair<decimal, Color>)))
+                return gradient.Markers.First().Value;
+            if (upper.Equals(default(KeyValuePair<decimal, Color>)))
+                return gradient.Markers.Last().Value;
+
+            // Interpolate
+            return InterpolateColor(lower, upper, (decimal)val);
+        }
+
+        /***************************************************/
+
+        [Description("Gets a color from a gradient at the spacified position scaled between from and to")]
+        public static Color Color(this Gradient gradient, double val, double from, double to)
+        {
+            return gradient.Color((val - from) / (to - from));
+        }
+
+        /***************************************************/
+        /****           Private Methods                 ****/
+        /***************************************************/
+
+        private static Color InterpolateColor(KeyValuePair<decimal, Color> lower, KeyValuePair<decimal, Color> upper, decimal val)
+        {
+            // safty check
+            val = val > lower.Key ? val < upper.Key ? val : upper.Key : lower.Key;
+
+            decimal diff = upper.Key - lower.Key;
+            decimal fraction = (val - lower.Key) / diff;
+
+            int A = (int)Math.Floor(Interpolate(lower.Value.A, upper.Value.A, fraction));
```

Comment:
Local variables should be lowercase, and I would personally prefer to see them spelt out in full to avoid confusion later down the line.